community/kernel/src/test/java/org/neo4j/kernel/impl/newapi/OperationsLockTest.java

```diff
@@ -429,6 +435,50 @@ public void shouldAcquireSchemaReadLockLazilyBeforeGettingAllConstraints()
         order.verify( locks ).acquireShared( LockTracer.NONE, ResourceTypes.RELATIONSHIP_TYPE, relTypeId );
     }
 
+    @Test
+    public void shouldAcquireSchemaWriteLockBeforeRemovingIndexRule() throws Exception
+    {
+        // given
+        SchemaIndexDescriptor index = SchemaIndexDescriptorFactory.forLabel( 0, 0 );
+        when( storeReadLayer.indexGetForSchema( any() )).thenReturn( index );
+
+        // when
+        operations.indexDrop( DefaultIndexReference.fromDescriptor( index ) );
+
+        // then
+        order.verify( locks ).acquireExclusive( LockTracer.NONE, ResourceTypes.LABEL, 0 );
+        order.verify( txState ).indexDoDrop( index );
+    }
+
+
+    @Test
+    public void shouldAcquireSchemaWriteLockBeforeCreatingUniquenessConstraint() throws Exception
+    {
+        // given
+        when( constraintIndexCreator.createUniquenessConstraintIndex( transaction, descriptor ) ).thenReturn( 42L );
+        when( storeReadLayer.constraintsGetForSchema(  descriptor.schema() ) ).thenReturn( Collections.emptyIterator() );
+        // when
+        operations.uniquePropertyConstraintCreate( descriptor );
+
+        // then
+        verify( locks ).acquireExclusive( LockTracer.NONE, ResourceTypes.LABEL, descriptor.getLabelId() );
```

Comment:
Should there be an `order.verify` here to see that the lock is taken before creation?